lib/xhr.js

```diff
@@ -182,11 +183,24 @@ function enableXHROverH2(XMLHttpRequest, configuration) {
                 this.__dispatchEvent(new ProgressEvent('progress'));
                 break;
             case XMLHttpRequest.DONE:
-                redefine(this, 'getResponseHeader', function(header) {
-                    return options.response.headers[header.toLowerCase()];
+                redefine(this, 'getResponseHeader', function (header) {
+                    var responseHeaders = options.response.headers;
+                    var trailers = options.response.trailers;
+                    for (var trailer in trailers) {
+                        if (trailers.hasOwnProperty(trailer)) {
+                            responseHeaders[trailer] = trailers[trailer];
+                        }
+                    }
+                    return responseHeaders[header.toLowerCase()];
                 });
-                redefine(this, 'getAllResponseHeaders', function() {
+                redefine(this, 'getAllResponseHeaders', function () {
                     var responseHeaders = options.response.headers;
+                    var trailers = options.response.trailers;
+                    for (var trailer in trailers) {
+                        if (trailers.hasOwnProperty(trailer)) {
+                            responseHeaders[trailer] = trailers[trailer];
+                        }
+                    }
                     return keys(responseHeaders).filter(function (responseHeader) {
```

Comment:
This is only my change. The rest is formatting issues. Sorry for the confusion.

Reply:
May be you want to split into two PRs (one for trailers and other for formatting).

Reply:
I reverted formatting changes.

Reply:
I don't really see the point of `redefine`. In any case, it appears to define a getter, meaning the function will be invoked every time `getResponseHeader` or `getAllResponseHeaders` is accessed. The `merge` will be computed every time as well, but I don't think headers or trailers should change once the request is "done". Can you find a way to compute the merged response headers only once?

Reply:
+1 on the argument that we should be able call `merge` only once. 

Also, why does it override `getResponseHeader` and `getResponseHeaders` on each XMLHttpRequest instance rather than on the prototype like is done elsewhere? `XMLHttpRequest.send` is a good example: https://github.com/akrambek/http2-cache.js/blob/develop/lib/xhr.js#L502. 

Regarding `redefine` in general,  it calls defineProperty with `value` in the descriptor (vs `get` or `set`. It's nitpicking, but it doesn't define a getter, per say.
```
var redefine = function (obj, prop, value) {

    if (obj[prop]) {
        // TODO, consider erasing scope/hiding (enumerable: false)
        obj["_" + prop] = obj[prop];
    }

    Object.defineProperty(obj, prop, {
        value: value,
        enumerable: obj.propertyIsEnumerable(prop),
        configurable: true
    });
};
```
 IMO, it seems like a reasonable utility to save a few lines of code if in order to override a method, which we do in several places.